bqplot/nbextension/GridHeatMap.js

```diff
@@ -121,6 +122,73 @@ define(["./components/d3/d3", "./Mark", "./utils"], function(d3, MarkViewModule,
             this.listenTo(this.model, "change:selected", this.update_selected);
             this.listenTo(this.model, "change:interactions", this.process_interactions);
         },
+        click_handler: function (args) {
+            var data = args.data;
+            var num_cols = this.model.colors[0].length;
+            var index = args.row_num * num_cols + args.column_num;
+            var row = args.row_num;
+            var column = args.column_num;
+            var that = this;
+            var idx = this.model.get("selected") ? utils.deepCopy(this.model.get("selected")) : [];
+            var selected = utils.deepCopy(this._cell_nums_from_indices(idx));
+            var elem_index = selected.indexOf(index);
+            // index of slice i. Checking if it is already present in the
+            // list
+            if(elem_index > -1 && d3.event.ctrlKey) {
+            // if the index is already selected and if ctrl key is
+            // pressed, remove the element from the list
+                idx.splice(elem_index, 1);
+            } else {
+                if(d3.event.shiftKey) {
+                    //If shift is pressed and the element is already
+                    //selected, do not do anything
+                    if(elem_index > -1) {
+                        return;
+                    }
+                    //Add elements before or after the index of the current
+                    //slice which has been clicked
+                    var row_index = (selected.length !== 0) ?
+                        idx[idx.length - 1][0] : row;
+                    var col_index = (selected.length !== 0) ?
+                        idx[idx.length - 1][1] : column;
+                    _.range(Math.min(row, row_index), Math.max(row, row_index)+1).forEach(function(i) {
+                        _.range(Math.min(column, col_index), Math.max(column, col_index)+1).forEach(function(j) {
+                            var cell_num = that._cell_nums_from_indices([[i, j]])[0];
+                            if (selected.indexOf(cell_num) === -1) {
+                                selected.push(cell_num);
+                                idx.push([i, j]);
+                            }
+                        })
+                    });
+                } else if(!(d3.event.ctrlKey)) {
```

Comment:
@rmenegaux if we move this to the top and make the next if statement an else. This will replicate Excels behaviour exactly, in that the `Shift` key will create a new block and the double `Shift` will have no effect. 

What does everyone else think?